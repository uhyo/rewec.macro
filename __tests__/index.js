const pluginTester = require("babel-plugin-tester").default;
const plugin = require("babel-plugin-macros");

const packageName = "rewec.macro"

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: {
    basic: `
      import { rec } from '../../${packageName}'

      console.log(rec(function sumTo(num){
        return num <= 0 ? 0 : num + sumTo(num - 1);
      }, 1e7));
    `,
    multipleReferencesInOneRec: `
      import { rec } from '../../${packageName}'

      console.log(rec(function fib(num){
        return num <= 1 ? num : fib(num - 1) + fib(num - 2)
      }, 1e7));
    `,
    multipleRecs: `
      import { rec } from '../../${packageName}'

      console.log(rec(function sumTo(num){
        return num <= 0 ? 0 : num + sumTo(num - 1);
      }, 1e7));
      console.log(rec(function fib(num){
        return num <= 1 ? num : fib(num - 1) + fib(num - 2)
      }, 1e7));
    `,
    nested: `
      import { rec } from '../../${packageName}'

      console.log(rec(function sumTo(num){
        const f = rec(function fib(num){
          return num <= 1 ? num : fib(num - 1) + fib(num - 2)
        }, 1e5);

        return num <= 0 ? 0 : num + sumTo(num - 1);
      }, 1e7));
    `,
  },
});
