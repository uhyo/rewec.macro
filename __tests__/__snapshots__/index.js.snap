// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin basic: basic 1`] = `

import { rec } from '../../rewec.macro'

console.log(rec(function sumTo(num){
  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

console.log(
  rec(function sumTo(num) {
    return num <= 0 ? 0 : num + sumTo(num - 1);
  }, 1e7)
);


`;

exports[`unknown plugin multipleRecs: multipleRecs 1`] = `

import { rec } from '../../rewec.macro'

console.log(rec(function sumTo(num){
  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));
console.log(rec(function fib(num){
  return num <= 1 ? num : fib(num - 1) + fib(num - 2)
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

console.log(
  rec(function sumTo(num) {
    return num <= 0 ? 0 : num + sumTo(num - 1);
  }, 1e7)
);
console.log(
  rec(function fib(num) {
    return num <= 1 ? num : fib(num - 1) + fib(num - 2);
  }, 1e7)
);


`;

exports[`unknown plugin multipleReferencesInOneRec: multipleReferencesInOneRec 1`] = `

import { rec } from '../../rewec.macro'

console.log(rec(function fib(num){
  return num <= 1 ? num : fib(num - 1) + fib(num - 2)
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

console.log(
  rec(function fib(num) {
    return num <= 1 ? num : fib(num - 1) + fib(num - 2);
  }, 1e7)
);


`;

exports[`unknown plugin nested: nested 1`] = `

import { rec } from '../../rewec.macro'

console.log(rec(function sumTo(num){
  const f = rec(function fib(num){
    return num <= 1 ? num : fib(num - 1) + fib(num - 2)
  }, 1e5);

  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

console.log(
  rec(function sumTo(num) {
    const f = rec(function fib(num) {
      return num <= 1 ? num : fib(num - 1) + fib(num - 2);
    }, 1e5);
    return num <= 0 ? 0 : num + sumTo(num - 1);
  }, 1e7)
);


`;
